strawberry/asgi/__init__.py

```diff
@@ -194,12 +194,19 @@ async def handle_async_results(
                 {"data": None, "errors": [format_graphql_error(error)]},
                 operation_id,
             )
+        except asyncio.CancelledError:
+            # CancelError will be received when the client disconnects and the handle_async_results task being cancelled
+            # Suppress the error and prevent contining to the sending completion message stage
+            return
 
         if (
             websocket.client_state != WebSocketState.DISCONNECTED
             and websocket.application_state != WebSocketState.DISCONNECTED
         ):
-            await self._send_message(websocket, GQL_COMPLETE, None, operation_id)
+            try:
+                await self._send_message(websocket, GQL_COMPLETE, None, operation_id)
+            except ConnectionClosedError:
+                pass
```

Comment:
I think we should catch ConnectionClosedError on L198 - L196 as well? 
There is a `except Exception as error` which might capture this error as well and try to send it as error to the client

Reply:
Just out of curiosity: where could `websockets.ConnectionClosedError` be raised?

I noticed `self._send_message` calls `starlette.WebSocket.send_json` calls `starlette.WebSocket.send`, but these methods appear to only raise `RuntimeError`s. (Finally `starlette.WebSocket.send` calls the `send` method passed to `strawberry.asgi.GraphQL.__call__`, thats where I lost track).